src/ui/zcl_abapgit_frontend_services.clas.abap

```diff
@@ -413,6 +424,16 @@ CLASS zcl_abapgit_frontend_services IMPLEMENTATION.
     ASSERT sy-subrc = 0.
     rv_path = ls_file_table-filename.
 
+    FIND FIRST OCCURRENCE OF REGEX '^/(.*/)?' IN rv_path MATCH LENGTH lv_len.
+    IF sy-subrc = 0.
+      gv_initial_folder = rv_path(lv_len).
+    ELSE.
+      FIND FIRST OCCURRENCE OF REGEX '^(.*\\)?' IN rv_path MATCH LENGTH lv_len.
+      IF sy-subrc = 0.
+        gv_initial_folder = rv_path(lv_len).
+      ENDIF.
+    ENDIF.
```

Comment:
perhaps extract this to a new method and add a few unit tests?

Reply:
done